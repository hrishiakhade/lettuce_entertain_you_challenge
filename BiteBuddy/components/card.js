import React from "react";
import { StyleSheet, Text, View, Pressable } from "react-native";
import { FONT_SIZES, COLORS, APP_STRINGS } from "../utils/constants";
import { RadioButton } from 'react-native-paper';


export const getQuestionText = (text) => {
    return (
        <Text style={styles.question}>{text}</Text>
    );
}

export const getButton = (isLastQuestion, handleNext, setOption, points) => {
    const buttonText = isLastQuestion ? APP_STRINGS.submit_button : APP_STRINGS.next_button;
    const buttonColor = points ? COLORS.secondary : COLORS.disabled;
    const buttonOpacity = points ? 1 : 0.6;

    const handlePress = () => {
        setOption();
        handleNext(points);
    };

    return (
        <Pressable
            style={({ pressed }) => [{
                opacity: pressed ? 0.9 : 1,
                backgroundColor: buttonColor,
                transform: [{ scale: pressed ? 0.95 : 1 }]
            }, styles.buttonStyle]}
            onPress={handlePress}
            disabled={!points}
        >
            <Text style={[styles.buttonText, { opacity: buttonOpacity }]}>
                {buttonText?.toUpperCase()}
            </Text>
        </Pressable>
    );
};



export const Cards = ({ questionData, isLastQuestion, handleNext }) => {
    const [option, setOption] = React.useState();

    return (
        <View style={styles.container}>
            {getQuestionText(questionData?.question?.toUpperCase())}
            <View style={styles.separator} />
            <RadioButton.Group onValueChange={value => setOption(value)} value={option}>
                {questionData?.options.map((option, index) => {
                    return (
                        <View key={index} style={{ flexDirection: 'row' }}>
                            <RadioButton.Item
                                label={option.text}
                                value={index}
                                position="leading"
                                labelStyle={styles.radioText}
                                color={COLORS.secondary}
                            />
                        </View>
                    );
                })}
            </RadioButton.Group>
            {getButton(isLastQuestion, handleNext, setOption, questionData?.options[option]?.points)}
        </View >
    );
}


const styles = StyleSheet.create({
    container: {
        marginTop: 30,
        borderWidth: 5,
        borderColor: COLORS.secondary,
        padding: 20,
        paddingBottom: 30,
        marginHorizontal: 20,
        minWidth: 300
    },
    question: {
        fontSize: FONT_SIZES.medium,
        fontWeight: "600",
        color: COLORS.primary,
        textAlign: 'center'
    },
    separator: {
        marginVertical: 20,
        height: 4,
        backgroundColor: COLORS.secondary
    },
    radioText: {
        fontSize: FONT_SIZES.medium,
        color: COLORS.primary,
        textAlign: 'left',
        marginLeft: 5
    },
    buttonStyle: {
        borderRadius: 20,
        paddingVertical: 10,
        paddingHorizontal: 40,
        position: 'absolute',
        bottom: -25,
        alignSelf: 'center',
        zIndex: 1
    },
    buttonText: {
        fontSize: FONT_SIZES.medium,
        fontWeight: 'bold',
        color: COLORS.primary,
        textAlign: 'center'
    }

});