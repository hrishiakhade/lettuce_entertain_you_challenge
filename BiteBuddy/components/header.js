import React from 'react';
import { StyleSheet, Text, View, ImageBackground, SafeAreaView } from 'react-native';
import { FONT_SIZES , COLORS } from '../utils/constants';

export const Header = ({ title, subtitle }) => {
    return (
        <View >
            <Text style={styles.title}>{title}</Text>
            <Text style={styles.subtitle}>{subtitle}</Text>
        </View >
    );
}

const styles = StyleSheet.create({
    title: {
        fontSize:FONT_SIZES.large,
        fontWeight: 'bold',
        color: COLORS.primary,
        textAlign: 'center',
    },
    subtitle: {
        fontSize: FONT_SIZES.medium,
        fontWeight: 'semibold',
        color: COLORS.primary,
        textAlign: 'center'
    }
});