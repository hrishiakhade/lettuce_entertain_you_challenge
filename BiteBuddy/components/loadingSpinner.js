
import { ActivityIndicator, StyleSheet, View ,Text} from 'react-native';
import { APP_STRINGS, COLORS, FONT_SIZES } from '../utils/constants';

export function LoadingSpinner(loading) {
    if (loading) {
        return (
            <View style={styles.container}>
                <ActivityIndicator size={'large'} color={COLORS.loading_indicator} style={styles.loadingStyle} />
                <Text style={styles.subText}>{APP_STRINGS.loading_text}</Text>
            </View>
        )
    }
    return null
}


const styles = StyleSheet.create({
    loadingStyle: {
        marginBottom: 15,
    },
    subText: {
        fontSize: FONT_SIZES.medium,
        color: COLORS.result_loading,
        textAlign: 'center',
        fontWeight: "600"
    },
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },
});
