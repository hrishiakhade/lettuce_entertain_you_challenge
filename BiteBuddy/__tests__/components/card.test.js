import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { Cards, getQuestionText, getButton } from '../../components/card';
import { RadioButton } from 'react-native-paper';


//  react-native-paper components are not fully compatible with react-native-testing-library


describe('Cards component', () => {
  test('renders question and options correctly', () => {
    const questionData = {
      question: 'Pick a Netflix show to binge watch',
      options: [
        { text: 'Stranger Things', points: 2 },
        { text: 'The Crown', points: 4 },
        { text: 'House of Cards', points: 1 },
        { text: 'Wednesday', points: 3 },
      ]
    };
    const isLastQuestion = false;
    const handleNext = jest.fn();

    const component = renderer.create(
      <Cards questionData={questionData} isLastQuestion={isLastQuestion} handleNext={handleNext} />
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('renders question text correctly', () => {
    const text = 'What is the first thing you reach for in the morning?';
    const questionText = getQuestionText(text);

    const component = renderer.create(questionText);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('renders next/submit button correctly', () => {
    const isLastQuestion = false;
    const handleNext = jest.fn();
    const setOption = jest.fn();
    const points = 3;

    const button = getButton(isLastQuestion, handleNext, setOption, points);

    const component = renderer.create(button);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });


  test('renders RadioButton correctly in stress test', () => {
    const questionData = {
      question: 'Sample question',
      options: [
        { text: 'Option 1', points: 1 },
        { text: 'Option 2', points: 2 },
        { text: 'Option 3', points: 3 },
      ],
    };

    const iterations = 100; // Number of iterations for stress testing

    for (let i = 0; i < iterations; i++) {
      const component = renderer.create(<Cards questionData={questionData} />);
      const radioButtons = component.root.findAllByType(RadioButton.Item);

      expect(radioButtons.length).toBe(questionData.options.length);

      questionData.options.forEach((option, index) => {
        expect(radioButtons[index].props.label).toBe(option.text);
        expect(radioButtons[index].props.value).toBe(index);
      });
    }
  });


  test('Radio Button selects the correct value stress test', () => {
    const questionData = {
      question: 'Sample question',
      options: [
        { text: 'Option 1', points: 1 },
        { text: 'Option 2', points: 2 },
        { text: 'Option 3', points: 3 },
      ],
    };

    const iterations = 100;

    for (let i = 0; i < iterations; i++) {
      let component;
      act(() => {
        component = renderer.create(<Cards questionData={questionData} />);
      });

      const radioGroup = component.root.findByType(RadioButton.Group);

      // Generate a random value within the options range
      const randomValue = Math.floor(Math.random() * questionData.options.length);

      act(() => {
        radioGroup.props.onValueChange(randomValue);
      });

      // Check if the selected value matches the randomly generated value
      expect(radioGroup.props.value).toBe(randomValue);
    }
  });
});
