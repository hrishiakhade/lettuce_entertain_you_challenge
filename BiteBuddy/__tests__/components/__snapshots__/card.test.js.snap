// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards component renders next/submit button correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "#9DB2BE",
        "opacity": 1,
        "transform": [
          {
            "scale": 1,
          },
        ],
      },
      {
        "alignSelf": "center",
        "borderRadius": 20,
        "bottom": -25,
        "paddingHorizontal": 40,
        "paddingVertical": 10,
        "position": "absolute",
        "zIndex": 1,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#262624",
          "fontSize": 19,
          "fontWeight": "bold",
          "textAlign": "center",
        },
        {
          "opacity": 1,
        },
      ]
    }
  >
    NEXT QUESTION
  </Text>
</View>
`;

exports[`Cards component renders question and options correctly 1`] = `
<View
  style={
    {
      "borderColor": "#9DB2BE",
      "borderWidth": 5,
      "marginHorizontal": 20,
      "marginTop": 30,
      "minWidth": 300,
      "padding": 20,
      "paddingBottom": 30,
    }
  }
>
  <Text
    style={
      {
        "color": "#262624",
        "fontSize": 19,
        "fontWeight": "600",
        "textAlign": "center",
      }
    }
  >
    PICK A NETFLIX SHOW TO BINGE WATCH
  </Text>
  <View
    style={
      {
        "backgroundColor": "#9DB2BE",
        "height": 4,
        "marginVertical": 20,
      }
    }
  />
  <View
    accessibilityRole="radiogroup"
  >
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityLabel="Stranger Things"
        accessibilityRole="radio"
        accessibilityState={
          {
            "busy": undefined,
            "checked": false,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            false,
            false,
            undefined,
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingHorizontal": 16,
                "paddingVertical": 8,
              },
              undefined,
            ]
          }
        >
          <View
            accessibilityLiveRegion="polite"
            accessibilityRole="radio"
            accessibilityState={
              {
                "busy": undefined,
                "checked": false,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            color="#9DB2BE"
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                false,
                {
                  "borderRadius": 18,
                },
              ]
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "borderColor": "rgba(73, 69, 79, 1)",
                  "borderRadius": 10,
                  "borderWidth": 2,
                  "height": 20,
                  "margin": 8,
                  "width": 20,
                }
              }
            />
          </View>
          <Text
            style={
              [
                {
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "letterSpacing": 0.15,
                  "lineHeight": 24,
                },
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "flexGrow": 1,
                    "flexShrink": 1,
                  },
                  false,
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "textAlign": "right",
                  },
                  {
                    "color": "#262624",
                    "fontSize": 19,
                    "marginLeft": 5,
                    "textAlign": "left",
                  },
                ],
              ]
            }
          >
            Stranger Things
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityLabel="The Crown"
        accessibilityRole="radio"
        accessibilityState={
          {
            "busy": undefined,
            "checked": false,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            false,
            false,
            undefined,
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingHorizontal": 16,
                "paddingVertical": 8,
              },
              undefined,
            ]
          }
        >
          <View
            accessibilityLiveRegion="polite"
            accessibilityRole="radio"
            accessibilityState={
              {
                "busy": undefined,
                "checked": false,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            color="#9DB2BE"
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                false,
                {
                  "borderRadius": 18,
                },
              ]
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "borderColor": "rgba(73, 69, 79, 1)",
                  "borderRadius": 10,
                  "borderWidth": 2,
                  "height": 20,
                  "margin": 8,
                  "width": 20,
                }
              }
            />
          </View>
          <Text
            style={
              [
                {
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "letterSpacing": 0.15,
                  "lineHeight": 24,
                },
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "flexGrow": 1,
                    "flexShrink": 1,
                  },
                  false,
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "textAlign": "right",
                  },
                  {
                    "color": "#262624",
                    "fontSize": 19,
                    "marginLeft": 5,
                    "textAlign": "left",
                  },
                ],
              ]
            }
          >
            The Crown
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityLabel="House of Cards"
        accessibilityRole="radio"
        accessibilityState={
          {
            "busy": undefined,
            "checked": false,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            false,
            false,
            undefined,
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingHorizontal": 16,
                "paddingVertical": 8,
              },
              undefined,
            ]
          }
        >
          <View
            accessibilityLiveRegion="polite"
            accessibilityRole="radio"
            accessibilityState={
              {
                "busy": undefined,
                "checked": false,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            color="#9DB2BE"
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                false,
                {
                  "borderRadius": 18,
                },
              ]
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "borderColor": "rgba(73, 69, 79, 1)",
                  "borderRadius": 10,
                  "borderWidth": 2,
                  "height": 20,
                  "margin": 8,
                  "width": 20,
                }
              }
            />
          </View>
          <Text
            style={
              [
                {
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "letterSpacing": 0.15,
                  "lineHeight": 24,
                },
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "flexGrow": 1,
                    "flexShrink": 1,
                  },
                  false,
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "textAlign": "right",
                  },
                  {
                    "color": "#262624",
                    "fontSize": 19,
                    "marginLeft": 5,
                    "textAlign": "left",
                  },
                ],
              ]
            }
          >
            House of Cards
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityLabel="Wednesday"
        accessibilityRole="radio"
        accessibilityState={
          {
            "busy": undefined,
            "checked": false,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            false,
            false,
            undefined,
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingHorizontal": 16,
                "paddingVertical": 8,
              },
              undefined,
            ]
          }
        >
          <View
            accessibilityLiveRegion="polite"
            accessibilityRole="radio"
            accessibilityState={
              {
                "busy": undefined,
                "checked": false,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            color="#9DB2BE"
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                false,
                {
                  "borderRadius": 18,
                },
              ]
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "borderColor": "rgba(73, 69, 79, 1)",
                  "borderRadius": 10,
                  "borderWidth": 2,
                  "height": 20,
                  "margin": 8,
                  "width": 20,
                }
              }
            />
          </View>
          <Text
            style={
              [
                {
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "letterSpacing": 0.15,
                  "lineHeight": 24,
                },
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "flexGrow": 1,
                    "flexShrink": 1,
                  },
                  false,
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "textAlign": "right",
                  },
                  {
                    "color": "#262624",
                    "fontSize": 19,
                    "marginLeft": 5,
                    "textAlign": "left",
                  },
                ],
              ]
            }
          >
            Wednesday
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "backgroundColor": "#ccc",
          "opacity": 1,
          "transform": [
            {
              "scale": 1,
            },
          ],
        },
        {
          "alignSelf": "center",
          "borderRadius": 20,
          "bottom": -25,
          "paddingHorizontal": 40,
          "paddingVertical": 10,
          "position": "absolute",
          "zIndex": 1,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#262624",
            "fontSize": 19,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          {
            "opacity": 0.6,
          },
        ]
      }
    >
      NEXT QUESTION
    </Text>
  </View>
</View>
`;

exports[`Cards component renders question text correctly 1`] = `
<Text
  style={
    {
      "color": "#262624",
      "fontSize": 19,
      "fontWeight": "600",
      "textAlign": "center",
    }
  }
>
  What is the first thing you reach for in the morning?
</Text>
`;
